src/platform-includes/sourcemaps/overview/javascript.mdx

```diff
@@ -1,39 +1,24 @@
-Integrate your project's source maps with Sentry using these steps:
+## Uploading Source Maps to Sentry
 
-### 1: Generate Source Maps
+We've compiled a list of guides on how to upload source maps to Sentry for the most popular JavaScript build tools:
 
-Source maps can be generated by transpilers, minifiers, or bundlers you might be using in your project. We provide instructions to configure common tools that output source maps [here](./generating/).
+- <PlatformLink to="/sourcemaps/uploading/webpack/">Webpack</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/typescript/">TypeScript (tsc)</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/rollup/">Rollup</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/vite/">Vite</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/esbuild/">esbuild</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/uglifyjs/">UglifyJS</PlatformLink>
+- <PlatformLink to="/sourcemaps/uploading/systemjs/">SystemJS</PlatformLink>
 
-### 2: Provide Source Maps to Sentry
+### Other Tools
 
-<PlatformSection notSupported={["node"]}>
-
-Source maps can be either:
-
-1.  Uploaded directly to Sentry (**strongly recommended**)
-2.  Served publicly over HTTP alongside your deployed code
-
-</PlatformSection>
-
-<PlatformSection notSupported={["javascript"]} supported={["node"]}>
-
-For Node.js projects, bundles and source maps need to be uploaded to Sentry.
-
-</PlatformSection>
-
-If you use [Webpack](https://webpack.js.org/) in your project, we recommend generating and uploading your source maps by way of [`sentry-webpack-plugin`](https://github.com/getsentry/sentry-webpack-plugin). If you use a different tool to generate source maps, you can use [`sentry-cli`](/product/cli/) to upload them to Sentry.
-
-Learn more about how to upload source maps [here](./uploading).
+If you're not using one of these tools, we assume you already know how to generate source maps for your toolchain and we recommend you upload them using the <PlatformLink to="/sourcemaps/uploading/cli/">Sentry CLI</PlatformLink>.
 
 <PlatformSection notSupported={["node"]}>
 
-If you host your source maps alongside your deployed code, make sure your code files end with a ``//# sourceMappingURL` comment pointing to the corresponding map.
-
-Learn more about publicly hosting source maps [here](./uploading/hosting-publicly/).
-
 <Note>
 
-By default, if Sentry can't find the uploaded files it needs, it will attempt to download them from the URLs in the stacktrace. To disable this, turn off "Enable JavaScript source fetching" in either your organization's "Security & Privacy" settings or your project's general settings.
+Though we strongly recommend uploading source maps as part of your build process, for browser applications it's also possible to <PlatformLink to="/sourcemaps/uploading/hosting-publicly/">host your source maps publicly</PlatformLink>.
```

Comment:
I wonder if we should explain why hosting is a bad idea (unless we do that somewhere else).

Reply:
I don't even know why we don't like this. Gonna leave it for now.

Reply:
Yup, point 3 is the big one, with the addition that if they do use hashes, they have to host all the old versions of the file/sourcemap forever.